compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs

```diff
@@ -113,9 +122,21 @@ impl ItemLikeVisitor<'v> for InherentOverlapChecker<'tcx> {
                 let ty_def_id = self.tcx.hir().local_def_id(item.hir_id);
                 let impls = self.tcx.inherent_impls(ty_def_id);
 
-                for (i, &impl1_def_id) in impls.iter().enumerate() {
-                    for &impl2_def_id in &impls[(i + 1)..] {
-                        if self.impls_have_common_items(impl1_def_id, impl2_def_id) {
+                // If there is only one inherent impl block,
+                // there is nothing to overlap check it with
+                if impls.len() <= 1 {
+                    return;
+                }
+
+                let impls_items = self.tcx.arena.alloc_from_iter(
```

Comment:
I'm quite certain that cloning the `impls.iter()` iterator would be cheaper than copying from an arena into probably the same arena once more :)